src/plugins/guided_onboarding/public/components/quit_guide_modal.tsx

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import React, { useState } from 'react';
+
+import {
+  EuiModal,
+  EuiModalBody,
+  EuiSpacer,
+  EuiTitle,
+  EuiText,
+  EuiModalFooter,
+  EuiButtonEmpty,
+  EuiButton,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import type { NotificationsSetup } from '@kbn/core-notifications-browser';
+import { GuideId } from '../../common/types';
+import { apiService } from '../services/api';
+
+interface QuitGuideModalProps {
+  closeModal: () => void;
+  currentGuide: GuideId;
+  notifications: NotificationsSetup;
+}
+
+export const QuitGuideModal = ({
+  closeModal,
+  currentGuide,
+  notifications,
+}: QuitGuideModalProps) => {
+  const [isDeleting, setIsDeleting] = useState<boolean>(false);
+
+  const deleteGuide = async () => {
+    setIsDeleting(true);
+    const { error } = await apiService.deleteGuide(currentGuide);
+
+    if (error) {
+      setIsDeleting(false);
+      notifications.toasts.addError(error, {
+        title: i18n.translate('guidedOnboarding.quitGuideModal.errorToastTitle', {
+          defaultMessage: 'There was an error quitting the guide. Please try again.',
```

Comment:
Do we want them to try again right away or should they wait a moment?

Reply:
Hmm... good question. I suppose it depends on the actual error. Do we have guidelines around error messages when an API request fails?

Reply:
The error message guidance that we've provided the platform (cloud) team in the past is that we want to A. tell them what happened, B. tell them what they can do about it, and C. only use please when we are asking for their patience. This message meets those guidelines, but it's just a matter of timing.